app/javascript/widget/store/modules/conversation/actions.js

```diff
@@ -1,112 +1,171 @@
-import {
-  createConversationAPI,
-  sendMessageAPI,
-  getMessagesAPI,
-  sendAttachmentAPI,
-  toggleTyping,
-  setUserLastSeenAt,
-} from 'widget/api/conversation';
-import { refreshActionCableConnector } from '../../../helpers/actionCable';
-
-import { createTemporaryMessage, getNonDeletedMessages } from './helpers';
+import ConversationAPI from 'widget/api/conversationPublic';
+import MessageAPI from 'widget/api/messagePublic';
+import { getNonDeletedMessages } from './helpers';
 
 export const actions = {
-  createConversation: async ({ commit, dispatch }, params) => {
-    commit('setConversationUIFlag', { isCreating: true });
+  fetchAllConversations: async ({ commit }) => {
     try {
-      const { data } = await createConversationAPI(params);
-      const {
-        contact: { pubsub_token: pubsubToken },
-        messages,
-      } = data;
-      const [message = {}] = messages;
-      commit('pushMessageToConversation', message);
-      refreshActionCableConnector(pubsubToken);
-      dispatch('conversationAttributes/getAttributes', {}, { root: true });
+      commit('setUIFlag', { isFetching: true });
+      const { data } = await ConversationAPI.get();
+      data.forEach(conversation => {
+        const { id: conversationId, messages } = conversation;
+        const { contact_last_seen_at: userLastSeenAt, status } = conversation;
+        const lastMessage = messages[messages.length - 1];
+        commit('addConversationEntry', conversation);
+        commit('addConversationId', conversationId);
+        commit('setConversationUIFlag', {
+          uiFlags: {},
+          conversationId,
+        });
+        commit('setConversationMeta', {
+          meta: { userLastSeenAt, status },
+          conversationId,
+        });
+        commit(
+          'messageV2/addMessagesEntry',
+          { conversationId, messages: [lastMessage] },
+          { root: true }
+        );
+        commit('addMessageIdsToConversation', {
+          conversationId,
+          messages: [lastMessage],
+        });
+      });
     } catch (error) {
-      // Ignore error
+      throw new Error(error);
```

Comment:
I think this throw inside a catch will be ignored, since it is not inside a try block.

Reply:
This will be thrown again, which an be caught when the action is called in UI. So we will know it has failed or not to show errors in UI.